src/hooks/useSmartAnimations.js

```
@@ -0,0 +1,162 @@
+      if ('getBattery' in navigator) {
+        navigator.getBattery().then((battery) => {
+          isLowBattery = battery.level < 0.2 && !battery.charging;
+        });
+      }
+      
+      // Calcul de la qualité d'animation
+      let quality = 'high';
+      if (reducedMotion || isLowEnd || isSlowNetwork || isLowBattery) {
+        quality = 'low';
+      } else if (isLowEnd) {
+        quality = 'medium';
```

Biome (2.1.2)

[error] This branch can never execute. Its condition is a duplicate or covered by previous conditions in the if-else-if chain.



(lint/suspicious/noDuplicateElseIf)